fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/physical/PhysicalHeapSort.java

```diff
@@ -71,6 +72,23 @@ public boolean hasLimit() {
         return limit > -1;
     }
 
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        PhysicalHeapSort that = (PhysicalHeapSort) o;
+        return offset == that.offset && limit == that.limit && Objects.equals(orderKeys, that.orderKeys);
```

Comment:
we need to discuss later, should we split PhysicalHeapSort to PhysicalHeapSort and PhysicalTopN